scripts/Muon/GUI/Common/plot_widget/plot_widget_view_interface.py

```diff
@@ -0,0 +1,128 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from abc import abstractmethod, ABCMeta
+from typing import List
+
+from qtpy import QtWidgets
+
+
+# This metaclass is required in order to implement the (strict) interface for the QtWidget
+class PlottingWidgetViewMeta(type(QtWidgets.QWidget), ABCMeta):
+    pass
+
+
+class PlotWidgetViewInterface(metaclass=PlottingWidgetViewMeta):
```

Comment:
in C++ we would call it `IPlotWidgetView`, I do like writing interface (took me ages to realise why we had the `I` at the start of the names.